include/exadg/acoustic_conservation_equations/user_interface/parameters.cpp

```diff
@@ -0,0 +1,167 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2023 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+// deal.II
+#include <deal.II/base/exceptions.h>
+
+// ExaDG
+#include <exadg/acoustic_conservation_equations/user_interface/parameters.h>
+#include <exadg/utilities/print_functions.h>
+
+namespace ExaDG
+{
+namespace Acoustics
+{
+// standard constructor that initializes parameters
+Parameters::Parameters()
+  : formulation(Formulation::Undefined),
+
+    // PHYSICAL QUANTITIES
+    start_time(0.),
+    end_time(-1.),
+    speed_of_sound(-1.),
+    density(-1.),
+
+    // TEMPORAL DISCRETIZATION
+    calculation_of_time_step_size(TimeStepCalculation::Undefined),
+    time_step_size(-1.),
+    max_number_of_time_steps(std::numeric_limits<unsigned int>::max()),
+    n_refine_time(0),
+    order_time_integrator(1),
+    start_with_low_order(true),
+    restarted_simulation(false),
+    adaptive_time_stepping(false),
+    restart_data(RestartData()),
+    solver_info_data(SolverInfoData()),
+
+    // SPATIAL DISCRETIZATION
+    grid(GridData()),
+    mapping_degree(1),
+    degree_u(1),
+    degree_p(1)
+{
+}
+
+void
+Parameters::check() const
+{
+  AssertThrow(formulation != Formulation::Undefined,
+              dealii::ExcMessage("parameter must be defined"));
+
+  // PHYSICAL QUANTITIES
+  AssertThrow(end_time > start_time, dealii::ExcMessage("parameter end_time must be defined"));
+  AssertThrow(speed_of_sound >= 0.0, dealii::ExcMessage("parameter must be defined"));
+  AssertThrow(density >= 0.0, dealii::ExcMessage("parameter must be defined"));
+
+  // TEMPORAL DISCRETIZATION
+  AssertThrow(calculation_of_time_step_size != TimeStepCalculation::Undefined,
+              dealii::ExcMessage("parameter must be defined"));
+
+  if(calculation_of_time_step_size == TimeStepCalculation::UserSpecified)
+    AssertThrow(time_step_size > 0., dealii::ExcMessage("parameter must be defined"));
+
+  if(adaptive_time_stepping)
+    AssertThrow(not(calculation_of_time_step_size == TimeStepCalculation::UserSpecified),
+                dealii::ExcMessage("parameter must be defined"));
```

Comment:
more precise - for now - could be `AsserThrow(not(adaptive_time_stepping)), ...)`.

Reply:
[2b35e7c](https://github.com/exadg/exadg/pull/592/commits/2b35e7c426c180de938d153403788ba3091e7767)